src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -87,131 +91,135 @@ public void jobLogic() {
-                        targetAddedSuccessfully = true;
+                boolean targetAddedSuccessfully = false;
+                long registerTargetStartTime = System.currentTimeMillis();
+                while (!targetAddedSuccessfully) {
+                    // Register target with target group.
+                    elbClient.registerTargets(registerTargetsRequest);
+                    waitAndCheckInstanceHealth("instance to register with ELB target group");
+                    // Check that the instance ID shows up in the health check.
+                    DescribeTargetHealthRequest healthRequest = new DescribeTargetHealthRequest()
+                        .withTargetGroupArn(otpServer.ec2Info.targetGroupArn);
+                    DescribeTargetHealthResult healthResult = elbClient.describeTargetHealth(healthRequest);
+                    for (TargetHealthDescription health : healthResult.getTargetHealthDescriptions()) {
+                        if (instance.getInstanceId().equals(health.getTarget().getId())) {
+                            LOG.info("Instance {} successfully added to target group!", instance.getInstanceId());
+                            targetAddedSuccessfully = true;
+                        }
+                    }
+                    // Wait for two minutes.
+                    if (taskHasTimedOut(registerTargetStartTime, 2 * 60 * 1000)) {
+                        failJob("Job timed out while checking for server bundle download status.");
+                        return;
                     }
                 }
-                // Wait for two minutes.
-                if (taskHasTimedOut(registerTargetStartTime, 2 * 60 * 1000)) {
-                    failJob("Job timed out while checking for server bundle download status.");
-                    return;
-                }
+                status.completeSuccessfully(
+                    String.format(
+                        "Server successfully registered with load balancer %s. OTP running at %s",
+                        otpServer.ec2Info.targetGroupArn,
+                        routerUrl
+                    )
+                );
+                LOG.info("View logs at {}", getUserDataLogS3Path());
+                deployJob.incrementCompletedServers();
+            } else {
+                failJob("There is no load balancer under which to register ec2 instance.");
             }
-            status.completeSuccessfully(
-                String.format(
-                    "Server successfully registered with load balancer %s. OTP running at %s",
-                    otpServer.ec2Info.targetGroupArn,
-                    routerUrl
-                )
-            );
-            LOG.info("View logs at {}", getUserDataLogS3Path());
-            deployJob.incrementCompletedServers();
-        } else {
-            failJob("There is no load balancer under which to register ec2 instance.");
```

Comment:
This else block should probably just move to the very top of the jobLogic. We should fail immediately before we even start up a server, right?